lib/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -34,32 +34,46 @@
 	 ************************************/
 
 	/**
-	 * Get the presence state for this Channel.
+	 * Get the presence state for this Channel. Implicitly attaches the
+	 * Channel. However, if the channel is in or moves to the FAILED
+	 * state before the operation succeeds, it will result in an error
 	 * @return: the current present members.
 	 * @throws AblyException
 	 */
-	public synchronized PresenceMessage[] get()  {
+	public synchronized PresenceMessage[] get() throws AblyException {
+		if (channel.state == ChannelState.failed) {
```

Comment:
My current focus is on Presence with this pull request. If other methods have features like  [RTP11b](http://docs.ably.io/client-lib-development-guide/features/#RTP11b), I am happy to implement that for them also